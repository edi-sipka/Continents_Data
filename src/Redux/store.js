/* eslint-disable */

import {
  configureStore,
  applyMiddleware,
  combineReducers,
} from '@reduxjs/toolkit';
import countryReducers from './reducers';
import thunk from 'redux-thunk';
import categorySelector from './category';

const rootReducer = combineReducers({
  countries: countryReducers,
  category: categorySelector,
});

const store = configureStore({ reducer: rootReducer }, applyMiddleware(thunk));
export default store;
