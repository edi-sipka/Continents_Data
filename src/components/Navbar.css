.navigation {
  background-color: rgba(114, 37, 93, 0.518);
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: 30px;
}

.heading {
  color: white;
  text-decoration: none;
  font-size: 24px;
  font-family: 'Lato', sans-serif;
  font-weight: 700;
}

.position {
  display: flex;
  gap: 20px;
  cursor: pointer;
  font-size: 20px;
}
