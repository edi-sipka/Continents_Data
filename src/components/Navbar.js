import React from 'react';
import { NavLink } from 'react-router-dom';
import {
  FaMicrophone,
  FaEllipsisH,
  FaArrowAltCircleLeft,
} from 'react-icons/fa';
import './Navbar.css';

function Navbar() {
  return (
    <div className="navigation">
      <NavLink to="/" element>
        <FaArrowAltCircleLeft className="btn" />
      </NavLink>
      <h2 className="heading">World Statistics</h2>

      <div className="position">
        <FaMicrophone />
        <FaEllipsisH />
      </div>
    </div>
  );
}

export default Navbar;
