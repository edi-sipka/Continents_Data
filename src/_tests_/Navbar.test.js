/* eslint-disable */

import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../Redux/store';
import Navbar from '../components/Navbar';

describe('Navbar rendering', () => {
  test('Navbar rendering', () => {
    const render = renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <Navbar />
        </BrowserRouter>
      </Provider>
    );
    expect(render).toMatchSnapshot();
  });
});
