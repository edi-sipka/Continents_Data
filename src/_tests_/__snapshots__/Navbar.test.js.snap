// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar rendering Navbar rendering 1`] = `
<div
  className="navigation"
>
  <a
    aria-current="page"
    className="active"
    element={true}
    href="/"
    onClick={[Function]}
  >
    <svg
      className="btn"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 512 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M256 504C119 504 8 393 8 256S119 8 256 8s248 111 248 248-111 248-248 248zm116-292H256v-70.9c0-10.7-13-16.1-20.5-8.5L121.2 247.5c-4.7 4.7-4.7 12.2 0 16.9l114.3 114.9c7.6 7.6 20.5 2.2 20.5-8.5V300h116c6.6 0 12-5.4 12-12v-64c0-6.6-5.4-12-12-12z"
      />
    </svg>
  </a>
  <h2
    className="heading"
  >
    World Statistics
  </h2>
  <div
    className="position"
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 352 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M176 352c53.02 0 96-42.98 96-96V96c0-53.02-42.98-96-96-96S80 42.98 80 96v160c0 53.02 42.98 96 96 96zm160-160h-16c-8.84 0-16 7.16-16 16v48c0 74.8-64.49 134.82-140.79 127.38C96.71 376.89 48 317.11 48 250.3V208c0-8.84-7.16-16-16-16H16c-8.84 0-16 7.16-16 16v40.16c0 89.64 63.97 169.55 152 181.69V464H96c-8.84 0-16 7.16-16 16v16c0 8.84 7.16 16 16 16h160c8.84 0 16-7.16 16-16v-16c0-8.84-7.16-16-16-16h-56v-33.77C285.71 418.47 352 344.9 352 256v-48c0-8.84-7.16-16-16-16z"
      />
    </svg>
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 512 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M328 256c0 39.8-32.2 72-72 72s-72-32.2-72-72 32.2-72 72-72 72 32.2 72 72zm104-72c-39.8 0-72 32.2-72 72s32.2 72 72 72 72-32.2 72-72-32.2-72-72-72zm-352 0c-39.8 0-72 32.2-72 72s32.2 72 72 72 72-32.2 72-72-32.2-72-72-72z"
      />
    </svg>
  </div>
</div>
`;
